testing/e2e/clients/dns.go

```diff
@@ -0,0 +1,207 @@
+	lgr.Info("starting to get dns")
+	defer lgr.Info("finished getting dns")
+
+	cred, err := GetAzCred()
+	if err != nil {
+		return nil, fmt.Errorf("getting az credentials: %w", err)
+	}
+
+	client, err := armdns.NewZonesClient(z.subscriptionId, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating client: %w", err)
+	}
+
+	resp, err := client.Get(ctx, z.resourceGroup, z.name, nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting dns: %w", err)
+	}
+
+	return &resp.Zone, nil
+}
+
+func (z *zone) GetName() string {
+	return z.name
+}
+
+func NewPrivateZone(ctx context.Context, subscriptionId, resourceGroup, name string, opts ...PrivateZoneOpt) (*privateZone, error) {
+	name = nonAlphanumericRegex.ReplaceAllString(name, "")
+	name = name + ".com"
+
+	lgr := logger.FromContext(ctx).With("name", name, "subscriptionId", subscriptionId, "resourceGroup", resourceGroup)
+	ctx = logger.WithContext(ctx, lgr)
+	lgr.Info("starting to create private zone")
+	defer lgr.Info("finished creating private zone")
+
+	cred, err := GetAzCred()
+	if err != nil {
+		return nil, fmt.Errorf("getting az credentials: %w", err)
+	}
+
+	client, err := armprivatedns.NewPrivateZonesClient(subscriptionId, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating client: %w", err)
+	}
+
+	pz := armprivatedns.PrivateZone{
+		Location: to.Ptr("global"),
+		Name:     to.Ptr(name),
+	}
+	for _, opt := range opts {
+		if err := opt(&pz); err != nil {
+			return nil, fmt.Errorf("applying private zone option: %w", err)
+		}
+	}
+	poller, err := client.BeginCreateOrUpdate(ctx, resourceGroup, name, pz, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating private zone: %w", err)
+	}
+
+	result, err := poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating private zone: %w", err)
+	}
+
+	return &privateZone{
+		name:           *result.PrivateZone.Name,
+		subscriptionId: subscriptionId,
+		resourceGroup:  resourceGroup,
+	}, nil
+}
+
+func (p *privateZone) GetName() string {
+	return p.name
+}
+
+func (p *privateZone) GetDnsZone(ctx context.Context) (*armprivatedns.PrivateZone, error) {
+	lgr := logger.FromContext(ctx).With("name", p.name, "subscriptionId", p.subscriptionId, "resourceGroup", p.resourceGroup)
+	ctx = logger.WithContext(ctx, lgr)
+	lgr.Info("starting to get private dns")
+	defer lgr.Info("finished getting private dns")
+
+	cred, err := GetAzCred()
+	if err != nil {
+		return nil, fmt.Errorf("getting az credentials: %w", err)
+	}
+
+	client, err := armprivatedns.NewPrivateZonesClient(p.subscriptionId, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("creating client: %w", err)
+	}
+
+	resp, err := client.Get(ctx, p.resourceGroup, p.name, nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting dns: %w", err)
+	}
+
+	return &resp.PrivateZone, nil
+}
+
+func (p *privateZone) LinkVnet(ctx context.Context, linkName, vnetId string) error {
+	linkName = nonAlphanumericRegex.ReplaceAllString(linkName, "")
```

Comment:
should we check that the length is greater than zero after removing the non-alpha chars with the regex?

Reply:
we can't realistically check that all name requirements are correct ourselves. the sdks will do that for us so it should be fine not to